base_layer/core/src/base_node/service/service.rs

```diff
@@ -342,7 +447,19 @@ where B: BlockchainBackend + 'static
                             .await
                             .map_err(|e| error!(target: LOG_TARGET, "Failed to ban peer: {:?}", e));
                     }
-                    error!(target: LOG_TARGET, "Failed to handle incoming block message: {}", e)
+                    error!(target: LOG_TARGET, "Failed to handle incoming block message: {}", e);
+                    if let Err(e) = network
+                        .report_gossip_message_validation_result(
+                            message_id,
+                            propagation_source,
+                            // Some other error occurred, we will not propagate, but we'll also not penalise the peer
+                            // for a failure not related to the message contents.
+                            gossipsub::MessageAcceptance::Ignore,
```

Comment:
following the above logic, should we also not reject on a. ban reason?

Reply:
Fair point, though I'm trying to avoid any issues with future gossiping. We handle the validation failure above. So this is not a result of validation failing, we're presumably banning the peer for some other reason which may not be related to the block in the gossip message e.g. IO error when requesting the block.

I would say that reducing their gossip score (which may result in them being removed from the mesh) because we could not connect to their RPC is not correct. Open to thoughts